src/contracts/middleware/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -0,0 +1,555 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+
+import "../interfaces/IBLSRegistryCoordinatorWithIndices.sol";
+import "../interfaces/ISocketUpdater.sol";
+import "../interfaces/IServiceManager.sol";
+import "../interfaces/IBLSPubkeyRegistry.sol";
+import "../interfaces/IVoteWeigher.sol";
+import "../interfaces/IStakeRegistry.sol";
+import "../interfaces/IIndexRegistry.sol";
+import "../interfaces/IPauserRegistry.sol";
+
+import "../libraries/EIP1271SignatureUtils.sol";
+import "../libraries/BitmapUtils.sol";
+import "../libraries/MiddlewareUtils.sol";
+
+import "../permissions/Pausable.sol";
+
+/**
+ * @title A `RegistryCoordinator` that has three registries:
+ *      1) a `StakeRegistry` that keeps track of operators' stakes
+ *      2) a `BLSPubkeyRegistry` that keeps track of operators' BLS public keys and aggregate BLS public keys for each quorum
+ *      3) an `IndexRegistry` that keeps track of an ordered list of operators for each quorum
+ * 
+ * @author Layr Labs, Inc.
+ */
+contract BLSRegistryCoordinatorWithIndices is EIP712, Initializable, IBLSRegistryCoordinatorWithIndices, ISocketUpdater, Pausable {
+    using BN254 for BN254.G1Point;
+
+    /// @notice The EIP-712 typehash for the `DelegationApproval` struct used by the contract
+    bytes32 public constant OPERATOR_CHURN_APPROVAL_TYPEHASH =
+        keccak256("OperatorChurnApproval(bytes32 registeringOperatorId, OperatorKickParam[] operatorKickParams)OperatorKickParam(address operator, BN254.G1Point pubkey, bytes32[] operatorIdsToSwap)BN254.G1Point(uint256 x, uint256 y)");
```

Comment:
the spacing looks slightly different than what I'm used to in typehash definitions. I think it's supposed to be
```
        keccak256("OperatorChurnApproval(bytes32 registeringOperatorId,OperatorKickParam[] operatorKickParams)OperatorKickParam(address operator,BN254.G1Point pubkey,bytes32[] operatorIdsToSwap)BN254.G1Point(uint256 x,uint256 y)");
```
I get this from looking at the EIP itself https://eips.ethereum.org/EIPS/eip-712 and in particular the example: "An example encoding is `Transaction(Person from,Person to,Asset tx)Asset(address token,uint256 amount)Person(address wallet,string name)`."

Reply:
resolved here https://github.com/Layr-Labs/eigenlayer-contracts/pull/35/commits/163c0189fbcf3a1a6bf3b9175d68df085cb61bec

Reply:
yes you're right! @0x0aa0 make sure you see this